server/services/MonitorService.js

```diff
@@ -4,77 +4,565 @@
+       if (Array.isArray(cleanMonitor.triggers)) {
+         cleanMonitor.triggers = cleanMonitor.triggers.map(({ id, last_triggered_time, ...trigger }) => trigger);
+       }
+       
+       cleanedBody = { ppl_monitor: cleanMonitor };
+     }
+
+     const resp = await client('transport.request', {
+       method: 'PUT',
+       path,
+       body: cleanedBody,
+       headers: DEFAULT_HEADERS,
+     });
+      return res.ok({ body: { ok: true, resp } });
+    } catch (err) {
+      console.error('Alerting - MonitorService - updatePPLMonitor:', err);
+      return res.ok({ body: { ok: false, resp: err.message } });
+    }
+  };
+
+  getPPLMonitor = async (context, req, res) => {
+    try {
+      const id = req.params.id;
+      const client = this.getClientBasedOnDataSource(context, req);
+      
+      // Use transport.request for MDS/AOSS compatibility
+      const raw = await client('transport.request', {
+        method: 'GET',
+        path: `/_plugins/_alerting/v2/monitors/${encodeURIComponent(id)}`,
+        headers: DEFAULT_HEADERS,
+      });
+      
+      const monitor =
+        _.get(raw, 'monitor_v2.ppl_monitor') ||
+        _.get(raw, 'ppl_monitor') ||
+        _.get(raw, 'monitor') ||
+        _.get(raw, '_source') ||
+        {};
+
+      // Normalize for UI safety
+      const normalized = {
+        ...monitor,
+        monitor_type: monitor.monitor_type || 'query_level',
+        item_type: monitor.workflow_type || monitor.monitor_type || 'query_level',
+        id,
+        version: _.get(raw, '_version', null),
+      };
+      normalized.triggers = Array.isArray(normalized.triggers) ? normalized.triggers : [];
+      normalized.ui_metadata = normalized.ui_metadata || { triggers: {} };
+
       return res.ok({
         body: {
           ok: true,
-          resp: createResponse,
+          resp: normalized,
+          version: _.get(raw, '_version', null),
+          ifSeqNo: _.get(raw, '_seq_no', null),
+          ifPrimaryTerm: _.get(raw, '_primary_term', null),
+          activeCount: 0,
+          dayCount: 0,
         },
       });
     } catch (err) {
-      console.error('Alerting - MonitorService - createMonitor:', err);
-      return res.ok({
-        body: {
-          ok: false,
-          resp: err.message,
-        },
+      console.error('Alerting - MonitorService - getPPLMonitor:', err);
+      return res.ok({ body: { ok: false, resp: err.message } });
+    }
+  };
+
+  deletePPLMonitor = async (context, req, res) => {
+    console.log("delete api called. req:", req);
+    try {
+      const id = req.params.id;
+      const client = this.getClientBasedOnDataSource(context, req);
+      
+      // Use transport.request for MDS/AOSS compatibility
+      const resp = await client('transport.request', {
+        method: 'DELETE',
+        path: `/_plugins/_alerting/v2/monitors/${encodeURIComponent(id)}`,
+        headers: DEFAULT_HEADERS,
+      });
+      
+      console.log("response:", resp);
```

Comment:
remove the console.logs and instead use logger if you need it (like important warnings, errors)